digital_benchmark/instagram_benchmark/views.py

```diff
@@ -120,6 +120,36 @@ def _load_media(self, dataProvider, instagram_parser, current_user, access_token
         instagram_parser.save_media_insight_data(
             all_user_media, current_user, access_token)
 
+class StartCrawlerView(APIView):
+   permission_classes = [IsAuthenticated]
+ 
+   def post(self, request):
+       app_user_id = request.user.id
+       unique_id = str(uuid4())
+       public_username = request.data.get('username')
+       setting = {
+           'unique_id': unique_id,
+           'USER_AGENT': settings.SCRAPER_AGENT
+       }
+       crawler_triggered = self._trigger_crawler(
+           setting, public_username, unique_id, app_user_id)
+       if crawler_triggered:
+           return Response({"Success": "Instagram crawler triggered from scrapy"})
+       return Response({"Error": "Error in triggering crawler"})
+ 
+   def _trigger_crawler(self, settings, public_username, unique_id, app_user_id):
+       task = scrapyd.schedule('default', 'insta_crawler', settings=settings,
+                               username=public_username, unique_id=unique_id, django_user_id=app_user_id)
+       crawler_stats = CrawlerStats()
+       crawler_stats.unique_id = unique_id
+       crawler_stats.task_id = task
+       crawler_stats.status = "Started"
+       try:
+           crawler_stats.save()
```

Comment:
No need to wrap this inside try catch. This is an overkill of cautionary measures.